packages/contracts/test/core/acl/integration-tests/shared-component.ts

```diff
@@ -0,0 +1,199 @@
+    await dao1.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+    await dao2.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+
+    // create a component
+    const TestSharedComponent = await ethers.getContractFactory(
+      'TestSharedComponent'
+    );
+    testComponent = await TestSharedComponent.deploy();
+    await testComponent.initialize(owningDao.address);
+
+    expectedACLAuthError = customError(
+      'ACLAuth',
+      testComponent.address,
+      testComponent.address,
+      ownerAddress,
+      ID_GATED_ACTION_ROLE
+    );
+  });
+
+  it('increments ids', async () => {
+    expect(await testComponent.callStatic.newObject(dao1.address)).to.be.equal(
+      0
+    );
+
+    const tx = await testComponent.newObject(dao1.address);
+    await tx.wait();
+    await ethers.provider.send('evm_mine', []);
+
+    expect(await testComponent.callStatic.newObject(dao1.address)).to.be.equal(
+      1
+    );
+  });
+
+  describe('idGatedAction:', async () => {
+    let oracle: TestIdGatingOracle;
+
+    beforeEach(async () => {});
+    it('executes if the id is allowed', async () => {
+      // deploy oracle and set allowed id
+      const allowedId = 0;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grants signers[0] the permission to do id gated actions with the deployed oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Create component with id 0
+      const tx = await testComponent.newObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      expect(testComponent.callStatic.idGatedAction(0));
+    });
+
+    it('reverts if the id is not allowed', async () => {
+      // deploy oracle and set allowed id
+      const allowedId = 1;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grant signers[0] the permission to do id gated actions with the deployed oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Create id-gated object associated with dao1
+      const tx = await testComponent.newObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      await expect(
+        testComponent.callStatic.idGatedAction(0)
+      ).to.be.revertedWith(expectedACLAuthError);
+    });
+
+    it('reverts if the role name is wrong', async () => {
+      // Deploy oracle and set allowed id
+      const allowedId = 1;
+
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grant signers[0] permission to do id gated actions with the deployed oracle
```

Comment:
Does this comment correspond to what the `WRONG_ROLE` does below?

Reply:
Oh, you are right. This was non-sense

Reply:
If the follow-up code available?